client/src/lib/browser.js

```diff
@@ -0,0 +1,27 @@
+import { useEffect, useState } from "react";
+
+export default function Browser() {
+  const [availableHeight, setAvailableHeight] = useState(calculateAvailableHeight());
+
+  useEffect(() => {
+    const handleWindowResize = () => {
+      setAvailableHeight(calculateAvailableHeight());
+    };
+
+    window.addEventListener("resize", handleWindowResize);
+
+    return () => {
+      window.removeEventListener("resize", handleWindowResize);
+    };
+  });
```

Comment:
Currently, this useEffect will run on every render of the component. We only need it to run once, after everything is loaded into the dom. To do this, we would add an empty array as a dependency to this useEffect.